lang/py/avro/schema.py

```diff
@@ -919,6 +984,10 @@ def to_json(self, names=None):
         to_dump['fields'] = [f.to_json(names) for f in self.fields]
         return to_dump
 
+    def validate(self, datum):
+        """Return self if datum is a valid representation of this schema, else None"""
+        return self if isinstance(datum, dict) and {f.name for f in self.fields}.issuperset(datum.keys()) else None
```

Comment:
Should this not validate the field values?

Reply:
Values are validated as children of the map schema.  Check the `_map_iterator` function in `io.py`.  That way, you get an exception that is specific to the field within the map that is wrong, not just an indication that there's a problem inside the map.

Reply:
Is this map schema? I thought it was record. Sorry! Very hard to tell in github mobile.

Reply:
it is, but for record the same thing happens.  Children of containers  validate their data types for themselves.